app-games/src/main/java/com/aptoide/android/aptoidegames/gamegenie/presentation/GameGenieViewModel.kt

```diff
@@ -0,0 +1,186 @@
+package com.aptoide.android.aptoidegames.gamegenie.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.aptoide.android.aptoidegames.gamegenie.domain.ChatInteraction
+import com.aptoide.android.aptoidegames.gamegenie.domain.GameGenieChat
+import com.aptoide.android.aptoidegames.gamegenie.domain.Token
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import javax.inject.Inject
+
+@HiltViewModel
+class GameGenieViewModel @Inject constructor(
+  private val gameGenieUseCase: GameGenieUseCase,
+) : ViewModel() {
+  private val viewModelState = MutableStateFlow(GameGenieViewModelState())
+
+  val uiState = viewModelState.map { it.toUiState() }
+    .stateIn(
+      viewModelScope,
+      SharingStarted.Eagerly,
+      viewModelState.value.toUiState()
+    )
+
+  init {
+    refreshToken()
+  }
+
+  private fun refreshToken() {
+    viewModelScope.launch {
+      gameGenieUseCase.getToken().let { token ->
+        viewModelState.update { it.copy(token = token) }
+      }
+    }
+  }
+
+  fun reload() {
+    viewModelScope.launch {
+      val chat = gameGenieUseCase.reloadConversation(
```

Comment:
i know the code style doesnt change it, but in this case it would be better for readability for it to be in the same line